### PR TITLE
add refund feature

### PR DESCRIPTION
**Pull request by Ductuan92**
add refund feature

### CODE REVIEW

File: src/Controller/Ticket/TicketController.php [hunk 1]
```diff
@@ -43,4 +44,13 @@ public function add(
...
         return $this->success([], Response::HTTP_CREATED);
     }
+
+    #[Route('/tickets/cancel/{id}', name: 'cancel', methods: 'POST')]
+    public function cancel(int $id, TicketRepository $ticketRepository, TicketService $ticketService)
+    {
+        $ticket = $ticketRepository->find($id);
+        $ticketService->cancel($ticket);
+
+        return $this->success([]);
```
Comment (tolehoai) [inline src/Controller/Ticket/TicketController.php:54]: Where is status code when it success?

File: src/Service/TicketService.php [hunk 2]
```diff
@@ -40,4 +47,21 @@ public function findAll(TicketRequest $ticketRequest)
...
+
+    public function cancel(Ticket $ticket): bool
+    {
+        $flight = $ticket->getFlight();
+        if (!$flight->isIsRefund() || $ticket->getCancelAt()) {
+            throw new Exception(ErrorsConstant::TICKET_NOT_REFUNDABLE);
+        }
+        $today = new DateTime();
+        $diff = $this->dateSubtract($today, $flight->getStartTime());
+        if ($this->secondToHours($diff) <  2) {
```
Comment (tolehoai) [inline src/Service/TicketService.php]: I think you should 2 in global variable

File: src/Controller/Ticket/TicketController.php (hunk 1 above)
Comment (Ductuan92) [reply src/Controller/Ticket/TicketController.php:54]: status code is auto set to 200 in success function

**Review by ntsanq — APPROVED** [review]

**Review by tolehoai — APPROVED** [review]
I think it can merge

**Pull request merged**